app/assets/javascripts/listings/ListingUnitService.js.coffee

```diff
@@ -29,49 +34,126 @@ ListingUnitService = ($http, ListingConstantsService, ListingIdentityService) ->
     else
       min
 
+  # FIXME: This might be broken by the new format
   Service.combineUnitSummaries = (listing) ->
-    # combined unitSummary is useful e.g. for overall occupancy levels across the whole listing
+    # Combined unitSummary is useful e.g. for overall occupancy levels across the whole listing.
     listing.unitSummaries ?= {}
     combined = _.concat(listing.unitSummaries.reserved, listing.unitSummaries.general)
     combined = _.omitBy(_.uniqBy(combined, 'unitType'), _.isNil)
-    # rename the unitType field to match how individual units are labeled
+    # Rename the unitType field to match how individual units are labeled.
     _.map(combined, (u) -> u.Unit_Type = u.unitType)
     Service._sortGroupedUnits(combined)
 
-  # Given ListingConstantsService.fieldsForUnitGrouping,
-  # combine similar units, and add up the number of available units
   Service._sumSimilarUnits = (units) ->
+    # Given ListingConstantsService.fieldsForUnitGrouping,
+    # combine similar units, and add up the number of available units.
     summaries = []
     # Create an identity function to group by unique price and income
     group = _.groupBy units, (unit) ->
       _.flatten(_.toPairs(_.pick(unit, ListingConstantsService.fieldsForUnitGrouping)))
     _.forEach group, (groupedUnits, id) ->
-      # Summarize each group by combining the unit details + # of units for that type/AMI combo
+      # Summarize each group by combining the unit details + # of units for that type/AMI combo.
       summary = _.pick(groupedUnits[0], ListingConstantsService.fieldsForUnitGrouping)
       summary.total = groupedUnits.length
       summaries.push(summary)
     summaries
 
-  # Group units by AMI % and Unit type
-  # Returns an object of the form:
-  #  {'50 (percent ami)':
-  #     {'1br': [{rent/price, # of units, etc}, ...], '2br': [...]},
-  #   '40 (percent ami)': {...}}
+  Service._getAMIAmount = (incomeList, occupancy) ->
+    if incomeList && incomeList.values?
```

Comment:
Not sure how best to handle cases where we can't find the AMI chart associated with the unit. This happens fairly rarely. Console.error should help it show up in Sentry, but I'm wondering if we should do something a bit more drastic so that it looks wrong for SMEs when they're previewing the listing.

Reply:
maybe we can check in on the sentry logs in a couple weeks to see if this ever happens? If it has happened we can think about error handling.

Unless you know for a fact that this will happen